test/strscan/test_stringscanner.rb

```diff
@@ -206,6 +206,17 @@ def test_pos_unicode
     assert_equal 11, s.charpos
   end
 
+  def test_charpos_not_use_byteslice
+    string = +'ruby'
+    scanner = create_string_scanner(string)
+    pre = Module.new do
+      def byteslice(*args)
+      end
+    end
+    string.singleton_class.prepend(pre)
+    assert_equal(0, scanner.charpos)
+  end
+
```

Comment:
I'm not sure this tests are really need for now?
It aims regression tests, but I feel a bit meaningless since https://github.com/ruby/strscan/pull/20#discussion_r602857935, https://github.com/ruby/strscan/commit/02bfa2c50fcdb41ed301b2e5867c35e382a2b0a8 😅

Reply:
How do you think this change?  https://github.com/ruby/strscan/pull/20/commits/1531d35cecac4b04e3b71df9565b88008b7a294d 🤔